src/Http/Routing/src/EndpointRoutingMiddleware.cs

```diff
@@ -87,7 +89,7 @@ static async Task AwaitMatch(EndpointRoutingMiddleware middleware, HttpContext h
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Curious, are async methods inlineable?

Reply:
Nope for lots of reasons.

Reply:
Feels like this method may as well be merged into the caller so it can be a single state machine.

Reply:
I'll benchmark it to see what makes sense. If it's 0 difference going to leave it as its own method since it's reused a lot

Reply:
Benchmarked the inlined version a few different times without resetting route data and got averages around:

| Description |       RPS | CPU (%) | Memory (MB) | Avg. Latency (ms) | Startup (ms) | First Request (ms) | Latency (ms) | Errors |
| ----------- | --------- | ------- | ----------- | ----------------- | ------------ | ------------------ | ------------ | ------ |
|             | 1,751,126 |      88 |          86 |              1.51 |          322 |              67.97 |         0.43 |      0 |

**Result**: 1.74% less RPS than baseline.

Going to leave as is.

Reply:
Kewl